torch/_dynamo/variables/builder.py

```diff
@@ -1498,6 +1500,135 @@ def wrap_symint(self, value):
 
         return unspec_var
 
+    def wrap_symfloat(self, value):
+        # SymFloat wrapping is special.  We first wrap it in the same way we
+        # do an unspecialized primitive, and then we item() it into a
+        # SymFloat.  Removal of the item() call is left to a later FX pass,
+        # mostly because that pass is more easily done after decomposition
+        # (Dynamo doesn't do decomposition right now).
+
+        if self.name in self.tx.output.unspec_variable_map:
+            return self.tx.output.unspec_variable_map[self.name]
+
+        # NB: we specialize on nan input, because our guard modeling in
+        # ShapeEnv cannot deal with nan
+        if (
+            torch._dynamo.config.specialize_float
+            or is_constant_source(self.get_source())
+            or math.isnan(value)
+        ):
+            self.install_guards(GuardBuilder.CONSTANT_MATCH)
+            return ConstantVariable.create(value=value, source=self.source)
+
+        # NB: We don't do automatic dynamic.  Since we have a guard mechanism,
```

Comment:
do you mean "we don't do static by default"? This code does automatic dynamic (which I think is good).

Reply:
Automatic dynamic specifically means "attempt it to be static the first time, then compile it dynamically the next time around". When `unspec_float = False`, I will actually *unconditionally* generate tensor inputs for all float inputs. An important optimization in this case is to remove these inputs if it turns out we do guard on the exact value. That's coming in another PR soon.

Reply:
right, it was just nomenclature. What you call automatic dynamic I call static by default